qiskit_nature/algorithms/initial_points/mp2_initial_point.py

```diff
@@ -0,0 +1,363 @@
+        self._ansatz = ansatz
+
+    @property
+    def threshold(self) -> float:
+        """The energy threshold for MP2 computation.
+
+        Computed initial point and energy corrections will be set to zero if their absolute value
+        is below this threshold.
+        """
+        return self._threshold
+
+    @threshold.setter
+    def threshold(self, threshold: float) -> None:
+        """The energy threshold for the MP2 computation.
+
+        Computed coefficients and energy corrections will be set to zero if their absolute value is
+        below this threshold.
+        """
+        try:
+            threshold = abs(float(threshold))
+        except TypeError:
+            threshold = 0.0
+
+        # Invalidate any previous computation.
+        self._corrections = None
+
+        self._threshold = float(threshold)
+
+    @property
+    def excitation_list(self) -> list[tuple[tuple[int, ...], tuple[int, ...]]]:
+        """The list of excitations."""
+        return self._excitation_list
+
+    @excitation_list.setter
+    def excitation_list(self, excitations: list[tuple[tuple[int, ...], tuple[int, ...]]]):
+        """The list of excitations.
+
+        Setting this will overwrite the excitation list from the ansatz.
+        """
+        # Invalidate any previous computation.
+        self._corrections = None
+
+        self._excitation_list = excitations
+
+    def to_numpy_array(self) -> np.ndarray:
+        """Convert the computed initial point to a numpy array.
+
+        Raises:
+            QiskitNatureError: If the initial point has not and cannot yet be computed.
+
+        Returns:
+            The initial point as a numpy array of MP2 correction coefficients.
+        """
+        if self._corrections is None:
+            if self._grouped_property is None or self._ansatz is None:
+                raise QiskitNatureError(self._missing_input_error_message)
+            self.compute()
+        return np.asarray([value[0] for value in self._corrections.values()])
+
+    def compute(
+        self,
+        grouped_property: GroupedSecondQuantizedProperty | None = None,
+        ansatz: UCC | None = None,
+    ) -> None:
+        """Compute the second-order Møller-Plesset perturbation theory (MP2) corrections.
+
+        This is intended for use as an initial point for the VQE parameters in combination with a UCC
+        ansatz.
+
+        :meth:`~compute` requires two-body molecular orbital electronic integrals and orbital
+        energies from
+        :class:`~qiskit_nature.properties.second_quantization.electronic.ElectronicEnergy`,
+        which should be passed in via the :attr:`grouped_property` attribute.
+
+        :meth:`compute` also requires the :attr:`excitation list` from the :attr:`~ansatz` to
+        ensure that the coefficients are mapped correctly in the initial point array. This can be
+        overwritten by setting :attr:`~excitation_list` directly.
```

Comment:
Some of this is duplicated from the class-level docstring. I think only one repetition on the class-level is fine so you can deduplicate things here.